In [1]:
import numpy as np 
import pandas as pd


In [3]:
heart = pd.read_csv('heart_data.csv')
heart

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0      0  18393       2     168    62.0    110     80            1   
1          1      1  20228       1     156    85.0    140     90            3   
2          2      2  18857       1     165    64.0    130     70            3   
3          3      3  17623       2     169    82.0    150    100            1   
4          4      4  17474       1     156    56.0    100     60            1   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69995  69995  99993  19240       2     168    76.0    120     80            1   
69996  69996  99995  22601       1     158   126.0    140     90            2   
69997  69997  99996  19066       2     183   105.0    180     90            3   
69998  69998  99998  22431       1     163    72.0    135     80            1   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69995     1      1     0       1       0  
69996     2      0     0       1       1  
69997     1      0     1       0       1  
69998     2      0     0       0       1  
69999     1      0     0       1       0  

[70000 rows x 14 columns]

ok so i have no idea why the age is so high (pls help for this)
and also im assuming we have to see whether or not an individual has high blood pressure and then see the factors that may affect it such as smoking and drinking alcohol 
and im cleaning the dataset in the sense where we only look at individuals of height between 160 and 175 cuz apparently avg singaporean height is 168 ish i think

In [5]:
heart_df = pd.DataFrame(heart)
heart_df

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0      0  18393       2     168    62.0    110     80            1   
1          1      1  20228       1     156    85.0    140     90            3   
2          2      2  18857       1     165    64.0    130     70            3   
3          3      3  17623       2     169    82.0    150    100            1   
4          4      4  17474       1     156    56.0    100     60            1   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69995  69995  99993  19240       2     168    76.0    120     80            1   
69996  69996  99995  22601       1     158   126.0    140     90            2   
69997  69997  99996  19066       2     183   105.0    180     90            3   
69998  69998  99998  22431       1     163    72.0    135     80            1   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
1         1      0     0       1       1  
2         1      0     0       0       1  
3         1      0     0       1       1  
4         1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69995     1      1     0       1       0  
69996     2      0     0       1       1  
69997     1      0     1       0       1  
69998     2      0     0       0       1  
69999     1      0     0       1       0  

[70000 rows x 14 columns]

In [30]:
height_1 = heart_df[165<heart_df['height']]
height_1

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0      0  18393       2     168    62.0    110     80            1   
3          3      3  17623       2     169    82.0    150    100            1   
7          7     12  22584       2     178    95.0    130     90            3   
10        10     15  22530       1     169    80.0    120     80            1   
11        11     16  18815       2     173    60.0    120     80            1   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69990  69990  99986  15094       1     168    72.0    110     70            1   
69993  69993  99991  19699       1     172    70.0    130     90            1   
69995  69995  99993  19240       2     168    76.0    120     80            1   
69997  69997  99996  19066       2     183   105.0    180     90            3   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
3         1      0     0       1       1  
7         3      0     0       1       1  
10        1      0     0       1       0  
11        1      0     0       1       0  
...     ...    ...   ...     ...     ...  
69990     1      0     0       1       1  
69993     1      0     0       1       1  
69995     1      1     0       1       0  
69997     1      0     1       0       1  
69999     1      0     0       1       0  

[29721 rows x 14 columns]

In [33]:
height_2 = height_1[height_1['height']<175]
height_2

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0      0  18393       2     168    62.0    110     80            1   
3          3      3  17623       2     169    82.0    150    100            1   
10        10     15  22530       1     169    80.0    120     80            1   
11        11     16  18815       2     173    60.0    120     80            1   
15        15     24  16782       2     172   112.0    120     80            1   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69984  69984  99975  17926       2     168    80.0    120     80            1   
69990  69990  99986  15094       1     168    72.0    110     70            1   
69993  69993  99991  19699       1     172    70.0    130     90            1   
69995  69995  99993  19240       2     168    76.0    120     80            1   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
3         1      0     0       1       1  
10        1      0     0       1       0  
11        1      0     0       1       0  
15        1      0     0       0       1  
...     ...    ...   ...     ...     ...  
69984     1      0     0       1       1  
69990     1      0     0       1       1  
69993     1      0     0       1       1  
69995     1      1     0       1       0  
69999     1      0     0       1       0  

[22348 rows x 14 columns]

ok so above is the dataframe for individuals within the height of 165 and 175.
now we clean again to find individuals within the weight of 60 to 90 kg.

In [34]:
weight_1 = height_2[height_2['weight']>60]
weight_1

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0      0  18393       2     168    62.0    110     80            1   
3          3      3  17623       2     169    82.0    150    100            1   
10        10     15  22530       1     169    80.0    120     80            1   
15        15     24  16782       2     172   112.0    120     80            1   
16        16     25  21296       1     170    75.0    130     70            1   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69984  69984  99975  17926       2     168    80.0    120     80            1   
69990  69990  99986  15094       1     168    72.0    110     70            1   
69993  69993  99991  19699       1     172    70.0    130     90            1   
69995  69995  99993  19240       2     168    76.0    120     80            1   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
3         1      0     0       1       1  
10        1      0     0       1       0  
15        1      0     0       0       1  
16        1      0     0       0       0  
...     ...    ...   ...     ...     ...  
69984     1      0     0       1       1  
69990     1      0     0       1       1  
69993     1      0     0       1       1  
69995     1      1     0       1       0  
69999     1      0     0       1       0  

[20895 rows x 14 columns]

In [36]:
weight_2 = weight_1[weight_1['weight']<90]
weight_2

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
0          0      0  18393       2     168    62.0    110     80            1   
3          3      3  17623       2     169    82.0    150    100            1   
10        10     15  22530       1     169    80.0    120     80            1   
16        16     25  21296       1     170    75.0    130     70            1   
24        24     35  16608       1     170    68.0    150     90            3   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69984  69984  99975  17926       2     168    80.0    120     80            1   
69990  69990  99986  15094       1     168    72.0    110     70            1   
69993  69993  99991  19699       1     172    70.0    130     90            1   
69995  69995  99993  19240       2     168    76.0    120     80            1   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
0         1      0     0       1       0  
3         1      0     0       1       1  
10        1      0     0       1       0  
16        1      0     0       0       0  
24        1      0     0       1       1  
...     ...    ...   ...     ...     ...  
69984     1      0     0       1       1  
69990     1      0     0       1       1  
69993     1      0     0       1       1  
69995     1      1     0       1       0  
69999     1      0     0       1       0  

[17713 rows x 14 columns]

ok still got alot of rows so need clean again
so we shall just look at males only which is gender 1 iinw
also im assuming that the age they give in days to make it an integer and converting it to years will make it become a float and idk how to do that so lets skip that.

In [39]:
final = weight_2[weight_2['gender'] == 1]
final

index     id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  \
10        10     15  22530       1     169    80.0    120     80            1   
16        16     25  21296       1     170    75.0    130     70            1   
24        24     35  16608       1     170    68.0    150     90            3   
31        31     43  23186       1     169    71.0    140     90            3   
49        49     64  16045       1     170    69.0    120     70            1   
...      ...    ...    ...     ...     ...     ...    ...    ...          ...   
69974  69974  99962  18226       1     168    75.0    120     80            1   
69977  69977  99965  16843       1     168    75.0    120     79            1   
69990  69990  99986  15094       1     168    72.0    110     70            1   
69993  69993  99991  19699       1     172    70.0    130     90            1   
69999  69999  99999  20540       1     170    72.0    120     80            2   

       gluc  smoke  alco  active  cardio  
10        1      0     0       1       0  
16        1      0     0       0       0  
24        1      0     0       1       1  
31        1      0     0       1       1  
49        1      0     0       1       0  
...     ...    ...   ...     ...     ...  
69974     1      0     0       1       0  
69977     1      0     0       1       0  
69990     1      0     0       1       1  
69993     1      0     0       1       1  
69999     1      0     0       1       0  

[8079 rows x 14 columns]